src/diem/testing/miniwallet/app/app.py

```diff
@@ -171,7 +184,50 @@ def _send_external_payment_txn(self, txn: Transaction) -> None:
         else:
             self._start_external_payment_txn(txn)
 
-    def _start_external_payment_txn(self, txn: Transaction) -> None:
+    def _start_external_payment_txn(self, txn: Transaction) -> None:  # noqa: C901
+        if identifier.diem_id.is_diem_id(str(txn.payee)):
+            reference_id = str(uuid.uuid4())
+            while True:
+                try:
+                    self.store.create(
+                        ReferenceID,
+                        before_create=self._validate_unique_reference_id,
+                        account_id=txn.account_id,
+                        reference_id=reference_id,
+                    )
+                    break
+                except ValueError:
+                    reference_id = str(uuid.uuid4())
+                    continue
```

Comment:
create a new function for above code: `_create_reference_id`, so here becomes: `reference_id = self._create_reference_id(...)`